From 16b7b284618d1652e694f6286f575ce82f5f03e5 Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Tue, 1 Sep 2015 16:50:20 +0800
Subject: [PATCH] ARM: dts: rk3288: add the interrupts property for PWM

We are assuming that PWM need use the property, we can support the
SPI interrupts for PWM.

At the moment, we can find the remotectl pwm is needed on box.
We can add the property for all PWMs. AFAIK, the pwm driver don't use it
but the drivers/input/remotectl/rockchip_pwm_remotectl.c

Change-Id: Ia223e6cc0e882477614b555bd09c86fd4bf8dbd7
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 arch/arm/boot/dts/rk3288.dtsi | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index ec1620797706..f4f921895f9d 100755
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -838,6 +838,9 @@
 	vop1pwm: pwm@ff9401a0 {
 		compatible = "rockchip,vop-pwm";
 		reg = <0xff9401a0 0x10>;
+
+		/* used by driver on remotectl'pwm */
+		interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
 		#pwm-cells = <2>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&vop1_pwm_pin>;
@@ -849,6 +852,9 @@
 	vop0pwm: pwm@ff9301a0 {
 		compatible = "rockchip,vop-pwm";
 		reg = <0xff9301a0 0x10>;
+
+		/* used by driver on remotectl'pwm */
+		interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
 		#pwm-cells = <2>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&vop0_pwm_pin>;
@@ -860,6 +866,9 @@
 	pwm0: pwm@ff680000 {
 		compatible = "rockchip,rk-pwm";
 		reg = <0xff680000 0x10>;
+
+		/* used by driver on remotectl'pwm */
+		interrupts = <GIC_SPI 74 IRQ_TYPE_LEVEL_HIGH>;
 		#pwm-cells = <2>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pwm0_pin>;
@@ -871,6 +880,9 @@
 	pwm1: pwm@ff680010 {
 		compatible = "rockchip,rk-pwm";
 		reg = <0xff680010 0x10>;
+
+		/* used by driver on remotectl'pwm */
+		interrupts = <GIC_SPI 75 IRQ_TYPE_LEVEL_HIGH>;
 		#pwm-cells = <2>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pwm1_pin>;
@@ -882,6 +894,9 @@
 	pwm2: pwm@ff680020 {
 		compatible = "rockchip,rk-pwm";
 		reg = <0xff680020 0x10>;
+
+		/* used by driver on remotectl'pwm */
+		interrupts = <GIC_SPI 76 IRQ_TYPE_LEVEL_HIGH>;
 		#pwm-cells = <2>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pwm2_pin>;
@@ -893,6 +908,9 @@
 	pwm3: pwm@ff680030 {
 		compatible = "rockchip,rk-pwm";
 		reg = <0xff680030 0x10>;
+
+		/* used by driver on remotectl'pwm */
+		interrupts = <GIC_SPI 77 IRQ_TYPE_LEVEL_HIGH>;
 		#pwm-cells = <2>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pwm3_pin>;
-- 
2.35.3

